tests/test_runtime.py

```diff
@@ -0,0 +1,68 @@
+import time
+from wandb.proto import wandb_internal_pb2 as pb
+import pytest
+
+
+def end_run_basic(interface):
+    time.sleep(3)
+
+
+def end_run_pause_resume(interface):
+    time.sleep(3)
+    interface.publish_pause()
+    time.sleep(3)
+    interface.publish_resume()
+    time.sleep(3)
+
+
+def end_run_pause_pause(interface):
+    time.sleep(3)
+    interface.publish_pause()
+    time.sleep(3)
+    interface.publish_pause()
+    time.sleep(3)
+
+
+def end_run_resume_resume(interface):
+    time.sleep(3)
+    interface.publish_resume()
+    time.sleep(3)
+    interface.publish_resume()
+    time.sleep(3)
+
+
+def end_run_resume_pause(interface):
+    time.sleep(3)
+    interface.publish_resume()
+    time.sleep(3)
+    interface.publish_pause()
+    time.sleep(3)
+
+
+def test_runtime_resume_point_2(
+    publish_util, test_settings, mock_server, start_run,
+):
+    mock_server.ctx["resume"] = True
+    test_settings.resume = "allow"
+
+    ctx_util = publish_util(
+        begin_cb=start_run, end_cb=end_run_basic, initial_start=True
```

Comment:
begin_cb=start_run not needed as it will be configured with initial_start=True right?